ts/Series/GeoHeatmap/GeoHeatmapSeries.ts

```diff
@@ -167,6 +272,22 @@ class GeoHeatmapSeries extends MapSeries {
 
     /* eslint-disable valid-jsdoc */
 
+    public init(): void {
+        const series = this;
+
+        super.init.apply(series, arguments);
+
+        if (
+            Object.keys(pick(series.hcEvents, {}))
+                .indexOf('afterDataClassLegendClick') === -1
+        ) {
```

Comment:
Instead of adding this event on the first init (and check if it is actually the first), I think it would be cleaner to add it to the class itself. See for example [how we do it wth the AreaRangeSeries](https://github.com/highcharts/highcharts/blob/46db0ddb81fc77e852fa1f1b19ccce1f04a7595a/ts/Series/AreaRange/AreaRangeSeries.ts#L649).